src/test/java/net/oauth/jsontoken/AsyncJsonTokenParserTest.java

```diff
@@ -150,23 +205,25 @@ private AsyncJsonTokenParser getAsyncJsonTokenParser(
     return new AsyncJsonTokenParser(clock, providers, executor, checkers);
   }
 
-  private <T extends Throwable> void assertFailsWithCauseAndErrorCode(
-      Class<T> throwableClass, ErrorCode errorCode, ThrowingRunnable runnable) {
-    ExecutionException e = assertThrows(ExecutionException.class, runnable);
-    assertTrue(throwableClass.isInstance(e.getCause()));
-
-    Throwable cause = e.getCause().getCause();
-    assertNotNull(cause);
-    assertTrue(InvalidJsonTokenException.class.isInstance(cause));
-    assertTrue(((InvalidJsonTokenException) cause).getErrorCode().equals(errorCode));
+  private InvalidJsonTokenException assertErrorCode(
+      ErrorCode errorCode, ThrowingRunnable runnable) {
+    InvalidJsonTokenException originalException =
+        assertThrows(InvalidJsonTokenException.class, runnable);
+    assertTrue(originalException.getErrorCode().equals(errorCode));
+    return originalException;
   }
 
-  private void assertFailsWithErrorCode(ErrorCode errorCode, ThrowingRunnable runnable) {
-    ExecutionException e = assertThrows(ExecutionException.class, runnable);
-    assertTrue(InvalidJsonTokenException.class.isInstance(e.getCause()));
+  private InvalidJsonTokenException assertFailsWithErrorCode(
```

Comment:
Can you just take the Future as input to this method? That way it's more clear how it's different from the one above.

Reply:
Changed.